cypress-io/cypress
# chore: add notification settings section

- Closes https://github.com/cypress-io/cypress/issues/26778

### Additional details


This PR adds a section to device settings called "Desktop notifications" that will allow users to configure which notifications they see. 

**NOTE** that these configuration values aren't used for anything yet. They will be used in https://github.com/cypress-io/cypress/issues/26788 to decide when to send notifications.

See the related issue for ClickUp requirements and Figma mockups

### Steps to test


1. Launch Cypress
2. Navigate to settings (gear icon in left nav bar)
3. Open device settings
4. Scroll down to Desktop notifications section
5. Click values in the form and verify that values are updated correctly
6. Refresh/reload the app and ensure that your selections were persisted

### How has the user experience changed?


<img width="1187" alt="Screenshot 2023-05-30 at 4 12 31 PM" src="https://github.com/cypress-io/cypress/assets/14275198/dd30859a-842c-498d-a50d-939fa1af04f6">

### PR Tasks


- [na] Has a PR for user-facing changes been opened in [`cypress-documentation`](https://github.com/cypress-io/cypress-documentation)? 
- [na] Have API changes been updated in the [`type definitions`](https://github.com/cypress-io/cypress/blob/develop/cli/types/cypress.d.ts)?
